src/components/NoteEditor/NoteEditor.jsx

```diff
@@ -0,0 +1,154 @@
+    createdAt: `${date.getDate()}/0${
+      date.getMonth() + 1
+    }/${date.getFullYear()}`,
+    timestamp: date.getTime(),
+  };
+  const {
+    user: { token },
+  } = useAuth();
+  const {
+    state: { notes },
+    setLoader,
+    dispatch,
+  } = useData();
+  const navigate = useNavigate();
+  const noteToDisplay = notes.filter((item) => item._id === id)[0];
+  const [note, setNote] = useState(id ? noteToDisplay : initVal);
+  const [expand, setExpand] = useState(false);
+
+  const activeNote = () => setExpand(true);
+
+  const hideNote = () => setExpand(false);
+
+  const handleChange = (e) => {
+    const name = e.target.name;
+    const value =
+      name === "tag" ? e.target.value.toLowerCase() : e.target.value;
+    setNote({ ...note, [name]: value });
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setLoader(true);
+    if (!id) {
+      const response = await postNotes({ token, note });
+      if (response.data.notes) {
+        dispatch({
+          type: ACTIONS.SetNotes,
+          payload: { notes: response.data.notes },
+        });
+      }
+    } else {
+      const editResponse = await editNotes({ notesId: id, note, token });
+      if (editResponse.data.notes) {
+        dispatch({
+          type: ACTIONS.SetNotes,
+          payload: { notes: editResponse.data.notes },
+        });
+        navigate("/notes");
+      }
+    }
+    setNote((prev) => ({
+      ...prev,
+      title: "",
+      tag: "",
+      isPinned: false,
+      timestamp: date.getTime(),
+    }));
+    setNote((prev) => ({ ...prev, body: "" }));
+    setLoader(false);
+  };
+  return (
+    <form
+      className="notes-form"
+      onSubmit={handleSubmit}
+      onDoubleClick={hideNote}
+    >
+      {expand && (
+        <div className="note-header">
+          <div className="input-grp">
+            <input
+              name="title"
+              className="input-title"
+              autoComplete="off"
+              value={note.title}
+              onChange={(e) => {
+                console.log(note.title);
+                handleChange(e);
+              }}
+              required
+              type="text"
+              placeholder="Title"
+            />
+          </div>
+          <span
+            className="pin-icon"
+            onClick={() =>
+              setNote((prev) => ({ ...prev, isPinned: !prev.isPinned }))
+            }
+          >
+            {note.isPinned ? <BsPinFill /> : <BsPin />}
+          </span>
+        </div>
+      )}
+      <QuillEditor
+        value={note.body}
+        setValue={(e) => setNote({ ...note, body: e })}
+        onFocus={activeNote}
+        isExpanded={expand}
+      />
+      {/* <textarea
+    className="input-area"
+    name="body"
+    required
+    autoComplete="off"
+    value={note.body}
+    onChange={handleChange}
+    placeholder={expand ? "Description" : "Click to add a note"}
+    onClick={activeNote}
+    onDoubleClick={hideNote}
+  /> */}
```

Comment:
You can remove the un-used comment here, If you meant to keep it for understanding then a descriptive comment will be more helpful, as you were adding comments in the previous files 👍️

Reply:
sure